controller/ingress_entry.go

```diff
@@ -6,9 +6,24 @@ import (
 	"time"
 )
 
-// IngressUpdate data
-type IngressUpdate struct {
-	Entries []IngressEntry
+// IngressEntries type
+type IngressEntries []IngressEntry
+
+// allows us to sort ingress entries.
+func (entries IngressEntries) Len() int {
+	return len(entries)
+}
+
+// allows us to sort ingress entries.
+func (entries IngressEntries) Less(i, j int) bool {
+	iEntry := fmt.Sprintf("%s:%s:%s:%s", entries[i].Namespace, entries[i].Name, entries[i].Host, entries[i].Path)
+	jEntry := fmt.Sprintf("%s:%s:%s:%s", entries[j].Namespace, entries[j].Name, entries[j].Host, entries[j].Path)
```

Comment:
Why do you think in ':' makes it not safe?  I think I actually don't even really need a delimiter actually.  And I don't think the number of ingress we are expecting even in the very very long term would make this a performance problem.